thrimbletrimmer/scripts/edit.js

```diff
@@ -1000,12 +1001,26 @@ async function initializeVideoInfo() {
 }
 
 function updateWaveform() {
+	const currentTime = DateTime.now().setZone("utc");
+
+	if (lastWaveformRequest && currentTime.diff(lastWaveformRequest).as("seconds") < 15) {
+		if (!waveformRequestTimeout) {
+			waveformRequestTimeout = setTimeout(updateWaveform, 15000);
+		}
+		return;
+	}
```

Comment:
nitpick: suppose an updateWaveform call comes in 14s after the previous. Then we will still wait a full 15s before executing it, for a total delay of 29s.
It would be more consistent to only delay for the difference in time, so the next update always happens 15s after the previous.


```suggestion
	if (lastWaveformRequest !== null) {
	    const remaining = 15 - currentTime.diff(lastWaveformRequest).as("seconds");
	    if (remaining > 0) {
			if (waveformRequestTimeout === null) {
				waveformRequestTimeout = setTimeout(updateWaveform, 1000 * remaining);
			}
			return;
		}
	}
```